processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -35,6 +37,8 @@ public TopNMapFn(
   {
     this.query = query;
     this.topNAlgorithm = topNAlgorithm;
+    this.dimension = query.getDimensionSpec().getDimension();
+    this.extractionFn = TopNQueryEngine.canApplyExtractionInPost(query) ? null : query.getDimensionSpec().getExtractionFn();
```

Comment:
Yup, which is another reason to push all that logic into `mergeResults()`, there's a bug that if you issue a query to the historical it can end up not doing some processing that we expect to be done on the broker...  Line 397 of ServerManager definitely does call `mergeResults` though.